### PR TITLE
Bug 1838215: Add support for RGW storage class

### PR DESCRIPTION
**Pull request by bipuladh**
- Update PVC filtering
- Update OBC filter

### CODE REVIEW

**Comment by afreen23** [comment]
We need to fix it for dashboards as well.

**Comment by bipuladh** [comment]
@afreen23 I see changes related to the dashboard to OB/C happening here https://github.com/openshift/console/pull/5341. Can we incorporate it there?

**Comment by cloudbehl** [comment]
@afreen23 This is not the scope of PR & also not planned yet. The priority is to fix the OBC creation for CEPH rgw & NooBaa First.

**Comment by bipuladh** [comment]
@afreen23 I just realized that the Object service dashboard will not only require those changes it will require changes in the queries as well. @anmolsachan I assume you're working on those parts right?

**Comment by afreen23** [comment]
> @afreen23 I see changes related to the dashboard to OB/C happening here #5341. Can we incorporate it there?

#5341 is the bug fix for some other issue, not specific to RGW.

> @afreen23 This is not the scope of PR & also not planned yet. The priority is to fix the OBC creation for CEPH rgw & NooBaa First.

I feel we need the changes on buckets card, otherwise if user visits dashboard he will not find the created OBCs with RGW. This is not a heavy change. 
You just need to do slight modification here:
https://github.com/openshift/console/blob/a4048cb8d8a910ffc538e55641d12fbdd1784df7/frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card.tsx#L46
The support should be consistent and touch all parts of the existing feature. 

Additionally metrics changes are surely required but thats some other use case.

**Comment by bipuladh** [comment]
@afreen23 metrics related to dashboard should happen all at one go. https://github.com/openshift/console/pull/5341 is specifically making changes to bucket cards. I am sure adding 1 more provisioner would not be an issue. However, I would prefer if we do all the query related changes together.

**Comment by afreen23** [comment]
> @afreen23 metrics related to dashboard should happen all at one go. #5341 is specifically making changes to bucket cards. I am sure adding 1 more provisioner would not be an issue. However, I would prefer it if we do all the query related changes together so that we don't miss out on some parts of the dashboard.

I checked on this and there is a problem with the OBC error status (that is not suffice from crd), and we might need some changes over there too.
My bad I missed on error counts. So doing it with query changes sounds good.

**Comment by bipuladh** [comment]
@afreen23 Just found out. Metrics change for the dashboard is moved out to 4.6.

**Comment by afreen23** [comment]
> @afreen23 Just found out. Metrics change for the dashboard is moved out to 4.6.

Thanks for the update, but for buckets card we decided to go generic. No change required as part of metrics.

**Comment by bipuladh** [comment]
> > @afreen23 Just found out. Metrics change for the dashboard is moved out to 4.6.
> 
> Thanks for the update, but for buckets card we decided to go generic. No change required as part of metrics.

I don't understand this. Why 1 card is generic and everything else is completely for MCG? This does not make sense. We should be consistent throughout one dashboard.

**Comment by afreen23** [comment]
> > > @afreen23 Just found out. Metrics change for the dashboard is moved out to 4.6.
> > 
> > 
> > Thanks for the update, but for buckets card we decided to go generic. No change required as part of metrics.
> 
> I don't understand this. Why 1 card is generic and everything else is completely for MCG? This does not make sense. We should be consistent throughout one dashboard.

OB/OBC is an API that any storage provider can use. Be it ceph RGW  or NooBaa Opertaor.
FYI , the dashboard activates when user installs MCG. Also the dashboard is not MCG dashboard but Object service dashboard, hence it must represent all object storage.
Hope this makes sense to you.

**Comment by bipuladh** [comment]
> > > > @afreen23 Just found out. Metrics change for the dashboard is moved out to 4.6.
> > > 
> > > 
> > > Thanks for the update, but for buckets card we decided to go generic. No change required as part of metrics.
> > 
> > 
> > I don't understand this. Why 1 card is generic and everything else is completely for MCG? This does not make sense. We should be consistent throughout one dashboard.
> 
> OB/OBC is an API that any storage provider can use. Be it ceph RGW or NooBaa Opertaor.
> FYI , the dashboard activates when user installs MCG. Also the dashboard is not MCG dashboard but Object service dashboard, hence it must represent all object storage.
> Hope this makes sense to you.

@afreen23 Sorry but I still do not get it. The dashboard charts and cards are all showing MCG details. My question is why only one card? And not the whole dashboard? This would definitely confuse a user. Sure the OB/C CRD is not MCG specific. But what about the rest? Even used storage is not MCG specific but we are not making any changes over there. Why these discrepancies? When was this decision taken to make the dashboard partially consistent? I would prefer if the whole dashboard be consistent than some aspects of it. @cloudbehl wdyt?

**Comment by cloudbehl** [comment]
@bipuladh @afreen23  The OB OBC page will show the OB & OBC created from both MCG & ceph RGW but MCG dashboard will not have any change in 4.5. The same counters will be shown as the dashboard is only for MCG & we will revamp it in the next release(as Bipul said).

File: frontend/packages/console-shared/src/utils/storage-utils.ts [hunk 1]
```diff
@@ -7,13 +7,13 @@ export const cephStorageProvisioners = [
...
 export const isCephProvisioner = (scProvisioner: string): boolean => {
   return cephStorageProvisioners.some((provisioner: string) =>
     _.endsWith(scProvisioner, provisioner),
   );
 };
 
-// To check that the storage class isn't noobaa based
-export const filterScOnProvisioner = (sc: StorageClass, provisioner: string = '') =>
-  sc.provisioner.includes(provisioner);
+export const isObjectSC = (sc: StorageClass) => objectStorageProvisioners.includes(sc.provisioner);
```
Comment (a2batic) [inline frontend/packages/console-shared/src/utils/storage-utils.ts:19]:
```suggestion
export const isObjectSC = (sc: StorageClass): boolean => {
  return objectStorageProvisioners.some((provisioner: string) =>
    sc.provisioner.includes(provisioner)
  );
};

```

Comment (bipuladh) [reply frontend/packages/console-shared/src/utils/storage-utils.ts:19]: This is not required if we modify the SC name to `openshift-storage.noobaa.io/obc`

**Pull request merged**